components/serialization/json/src/main/java/com/microsoft/kiota/serialization/JsonParseNode.java

```diff
@@ -125,69 +158,12 @@ public JsonParseNode(@Nonnull final JsonElement node) {
-                                                    this.setOnAfterAssignFieldValues(onAfter);
-                                                }
-                                            };
-                                    if (targetClass == Boolean.class) {
-                                        return (T) itemNode.getBooleanValue();
-                                    } else if (targetClass == Short.class) {
-                                        return (T) itemNode.getShortValue();
-                                    } else if (targetClass == Byte.class) {
-                                        return (T) itemNode.getByteValue();
-                                    } else if (targetClass == BigDecimal.class) {
-                                        return (T) itemNode.getBigDecimalValue();
-                                    } else if (targetClass == String.class) {
-                                        return (T) itemNode.getStringValue();
-                                    } else if (targetClass == Integer.class) {
-                                        return (T) itemNode.getIntegerValue();
-                                    } else if (targetClass == Float.class) {
-                                        return (T) itemNode.getFloatValue();
-                                    } else if (targetClass == Long.class) {
-                                        return (T) itemNode.getLongValue();
-                                    } else if (targetClass == UUID.class) {
-                                        return (T) itemNode.getUUIDValue();
-                                    } else if (targetClass == OffsetDateTime.class) {
-                                        return (T) itemNode.getOffsetDateTimeValue();
-                                    } else if (targetClass == LocalDate.class) {
-                                        return (T) itemNode.getLocalDateValue();
-                                    } else if (targetClass == LocalTime.class) {
-                                        return (T) itemNode.getLocalTimeValue();
-                                    } else if (targetClass == PeriodAndDuration.class) {
-                                        return (T) itemNode.getPeriodAndDurationValue();
-                                    } else {
-                                        throw new RuntimeException(
-                                                "unknown type to deserialize "
-                                                        + targetClass.getName());
-                                    }
-                                }
-                            };
-                        }
-                    };
-
-            for (T elem : iterable) {
-                result.add(elem);
+            while (sourceIterator.hasNext()) {
+                JsonElement item = sourceIterator.next();
```

Comment:
we have those 4 first lines at 4 different places, do you think we could use a method to get the JsonParseNode from a JsonElement and set the the before/after assign field values callbacks?

Reply:
no strong opinion here, I don't know how much those routines are expected to change and, possibly, diverge over time.

Reply:
I'd say not at all, besides the last line which is different for each block

Reply:
ok, let me refactor this too

Reply:
refactored the iteration on the Array, I noticed only after that's slight different from what you asked for, but I think it reads pretty well now ... let me know